pkg/gofr/datasource/pubsub/google/health.go

```diff
@@ -0,0 +1,89 @@
+package google
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"gofr.dev/pkg/gofr/datasource"
+	"google.golang.org/api/iterator"
+)
+
+func (g *googleClient) Health() (health datasource.Health) {
+	health.Details = make(map[string]interface{})
+
+	health.Status = datasource.StatusUp
+	health.Details["projectID"] = g.Config.ProjectID
+	health.Details["backend"] = "GOOGLE"
+
+	health.Status, health.Details["writers"] = g.getWriterDetails()
+	health.Status, health.Details["readers"] = g.getReaderDetails()
```

Comment:
health.Status should be an OR of writers and readers, in the current scenario is writer is not working fine but readers are, status would be UP